transport/tchannel/handler.go

```diff
@@ -108,33 +121,39 @@ func (h handler) handle(ctx context.Context, call inboundCall) {
 		call.Response().Blackhole()
 		return
 	}
-	if err != nil && !responseWriter.isApplicationError {
-		// TODO: log error
-		_ = call.Response().SendSystemError(getSystemError(err))
+	if err != nil && !responseWriter.IsApplicationError() {
+		sendErr := call.Response().SendSystemError(getSystemError(err))
+		if sendErr != nil {
+			h.logger.Error("SendSystemError failed", zap.Error(sendErr))
+		}
+		h.logger.Error("tchannel transport handler request failed", zap.Error(err))
```

Comment:
nit: A slightly cleaner message would be "SendSystemError failed" and "handler failed" respectfully. Zap's loggers can add [a name to its logger](https://godoc.org/go.uber.org/zap#Logger.Named). Doing so would give the user more context without repeating yourself.